controllers/repository_controller.go

```diff
@@ -18,24 +18,44 @@ package controllers
 
 import (
 	"context"
+	"time"
 
+	"github.com/go-logr/logr"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	corev1alpha1 "github.com/kubebb/core/api/v1alpha1"
+	"github.com/kubebb/core/pkg/repository"
+	"github.com/kubebb/core/pkg/utils"
 )
 
+// Fixed, no lower configuration allowed
+const minIntervalSeconds = 300
+
 // RepositoryReconciler reconciles a Repository object
 type RepositoryReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
+
+	Ctx      context.Context
```

Comment:
Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. 
see https://pkg.go.dev/context or https://go.dev/blog/context-and-structs